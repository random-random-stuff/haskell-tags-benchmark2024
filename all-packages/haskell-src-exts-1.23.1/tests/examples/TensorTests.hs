{-# LANGUAGE ConstraintKinds, FlexibleContexts, DataKinds, NoImplicitPrelude,
             RebindableSyntax, ScopedTypeVariables, TypeFamilies, TypeOperators,
             UndecidableInstances #-}

module TensorTests (tensorTests) where

type TMRParams = ( '(,) <$> Tensors) <*> MRCombos
tmrParams :: Proxy TMRParams
tmrParams = Proxy

--type ExtParams = ( '(,) <$> Tensors) <*> MRExtCombos
type TrEmParams = ( '(,) <$> Tensors) <*> MM'RCombos
tremParams :: Proxy TrEmParams
tremParams = Proxy

type NormParams = ( '(,) <$> '[RT]) <*> (Filter Liftable MRCombos)

--data Liftable :: TyFun (Factored, *) Bool -> *
type instance Apply Liftable '(m,zq) = Int64 :== (LiftOf zq)

