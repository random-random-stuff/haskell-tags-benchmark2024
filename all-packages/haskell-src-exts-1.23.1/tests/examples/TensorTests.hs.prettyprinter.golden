{-# LANGUAGE ConstraintKinds, FlexibleContexts, DataKinds,
  NoImplicitPrelude, RebindableSyntax, ScopedTypeVariables,
  TypeFamilies, TypeOperators, UndecidableInstances #-}
module TensorTests (tensorTests) where

type TMRParams = ('(,) <$> Tensors) <*> MRCombos

tmrParams :: Proxy TMRParams
tmrParams = Proxy

type TrEmParams = ('(,) <$> Tensors) <*> MM'RCombos

tremParams :: Proxy TrEmParams
tremParams = Proxy

type NormParams = ('(,) <$> '[RT]) <*> (Filter Liftable MRCombos)

type instance Apply Liftable '(m, zq) = Int64 :== (LiftOf zq)
